src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java

```diff
@@ -279,49 +279,70 @@ public static void scan(Run build, PrintStream scanLog) {
      * @param buildNum - Build object
      * @param buildUrl - Full URL of build
      * @param scanLog - PrintStream for the build log
-     * @return boolean true if message successfully created, false otherwise
+     * @return String Slack message with failure name, category and description if message successfully created, null otherwise
      */
-    public static boolean createSlackMessage(List<FoundFailureCause> foundCauseList,
+    public static String createSlackMessage(List<FoundFailureCause> foundCauseList,
             boolean notifySlackOfAllFailures, List<String> slackFailureCauseCategories,
             String buildName, String buildNum, String buildUrl, PrintStream scanLog) {
         boolean notifySlackOfFailure = false;
         StringBuilder bufBuildFailCause = new StringBuilder();
 
         /* Check if one of the failure causes for the build matches those specified in plugin's slack settings. */
         for (FoundFailureCause foundCause : foundCauseList) {
-            if (!notifySlackOfAllFailures) {
+            if (notifySlackOfAllFailures) {
+                //Add two new lines between found categories
+                if (bufBuildFailCause.length() != 0) {
+                    bufBuildFailCause.append("\n\n");
+                }
+                //Create list for slack message with failure Name, Cegory and Description from build
+                bufBuildFailCause.append("*Failure Name:* ");
+                bufBuildFailCause.append(foundCause.getName());
+                bufBuildFailCause.append("\n");
+                bufBuildFailCause.append("*Failure Categories:* ");
+                bufBuildFailCause.append(foundCause.getCategories());
+                bufBuildFailCause.append("\n");
+                bufBuildFailCause.append("*Description:* ");
+                bufBuildFailCause.append(foundCause.getDescription());
+            } else {
+                // Only notify the selected categories even if others occur
                 List<String> categories = foundCause.getCategories();
                 if (categories != null) {
                     for (String category : categories) {
                         if (failureCategoryMatches(category, slackFailureCauseCategories)) {
                             notifySlackOfFailure = true;
-                            break;
+                            //Add two new lines between found categoreis
+                            if (bufBuildFailCause.length() != 0) {
+                                bufBuildFailCause.append("\n\n");
+                            }
+                            // Create list for slack message with failure Name, Cegory and Description from build
+                            bufBuildFailCause.append("*Failure Name:* ");
```

Comment:
This looks weird. If a failureCause has multiple categories you will print all it's information for each matching category.

Reply:
No it will print both Categories per Failure Cause:
for example:
Job "fail-exit1-test" build #373 FAILED due to following failure causes:
Failure Name: Fastlane Build hang
Failure Categories: [Fastlane, :apple:]
Description: fastlane finished with errors

Reply:
I have added a message to display how it looks in slack in the description